src/storage-controller/src/lib.rs

```diff
@@ -1786,6 +1795,10 @@ where
 
             Some(m) = self.internal_response_queue.recv() => Some(m),
             Some(m) = instance_responses.next() => m,
+            _ = self.maintenance_ticker.tick() => {
+                self.maintenance_scheduled = true;
```

Comment:
Is there any benefit from maintaining this flag and doing the work in `process`? My understanding is that just doing the work here will have the same effect since whoever calls ready will call process anyway if we resolve and `maintain` is a sync method

Reply:
Makes sense, I don't have a strong opinion, just wanted to raise this question. LGTM!